### PR TITLE
Email address and Comments

### PR DESCRIPTION
**Pull request by cxfloeder**
Store the user's email when they write comments, and print the address when displaying the comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -101,13 +111,10 @@ private String messageListAsJson(List<Map<String, String>> commentList) {
             {
                 break;
             }
-            String comment = (String) entity.getProperty(COMMENT_INPUT);
-            if(!comment.equals(""))
+            Comment comObject = new Comment(entity);
```
Comment (ianjjohnson) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:114]: comObject can just be named "comment"

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -27,25 +27,38 @@
...
+        response.setContentType(TEXT_TYPE);
         PrintWriter out = response.getWriter();
         UserService userService = UserServiceFactory.getUserService();
-
+            
         if(userService.isUserLoggedIn()) {
             String userEmail = userService.getCurrentUser().getEmail();
             String logoutURL = userService.createLogoutURL(LOG_PAGE_URL);
 
+//            response.sendRedirect(COMMENTS_URL);
```
Comment (ianjjohnson) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:44]: remove commented-out line of code

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public Comment(HttpServletRequest request, String userEmail) {
+        timestamp = System.currentTimeMillis();
+        message = request.getParameter(COMMENT_FORM_ID); 
+        email = userEmail;
+    }
+
+    /** Return an Entity for DataStore */
+    public Entity toEntity() {
```
Comment (ianjjohnson) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:29]: nice! :-)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -69,6 +69,9 @@ function validatePassword(form) {
     }
 }
 
+function isLoggedIn() {
```
Comment (ianjjohnson) [inline portfolio/src/main/webapp/script.js:72]: I would either implement this or remove it.
(Or at the very least add a comment and a fake implementation like "return false")

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 5]
```diff
@@ -27,25 +27,38 @@
...
             out.println("<p>Hello " + userEmail + "! You are logged in!</p>");
-            out.println("<p>Logout <a href=\"" + logoutURL + "\">here</a>.</p>");
+            out.println("<p>To view the comments, click <a href=\"" + COMMENTS_URL + "\">here</a>.</p>");
```
Comment (ianjjohnson) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:46]: I'm fine leaving this as-is, but just so you know, a more common paradigm is to not include HTML in your backend and only send the data itself to the UI, letting the Javascript/HTML handle actually creating the proper HTML elements.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (cxfloeder) [reply portfolio/src/main/webapp/script.js:72]: I am planning on implementing it in the next step.

**Pull request merged**